corehq/apps/cloudcare/views.py

```diff
@@ -410,12 +410,21 @@ def put(self, request, *args, **kwargs):
         body = json.loads(request.body.decode('utf-8'))
         access = get_application_access_for_domain(self.domain)
         access.restrict = body['restrict']
+        disable_ff = body['disable_feature_flag']
         access.sqlappgroup_set.all().delete()
         access.sqlappgroup_set.set([
             SQLAppGroup(app_id=app_group['app_id'], group_id=app_group.get('group_id'))
             for app_group in body['app_groups']
         ], bulk=False)
         access.save()
+        if disable_ff and not access.restrict:
+            toggles.WEB_APPS_PERMISSIONS_VIA_GROUPS.set(self.domain, False, namespace=toggles.NAMESPACE_DOMAIN)
+            # This view is not accessible after the FF is disabled
+            redirect_url = reverse('users_default', args=[self.domain])
```

Comment:
should this direct them to the roles and permissions page instead?

Reply:
Yes that would be better [1eed3e8](https://github.com/dimagi/commcare-hq/pull/35245/commits/1eed3e8a6c1d3c7c21094f6b949c9bb459349a12)